library/cwm/src/lib/cwm/abstract_widget.rb

```diff
@@ -1,6 +1,22 @@
 require "abstract_method"
 
 module CWM
+  # A Yast::Term that can be passed as is to Yast::UI methods
+  # (OpenDialog, ReplaceWidget)
+  class UITerm < Yast::Term; end
+
+  # A Yast::Term that contains instances of {CWM::AbstractWidget}.
+  # Can be passed to {Yast::CWMClass#show Yast::CWM.show}
+  class WidgetTerm < Yast::Term; end
+
+  # A Yast::Term that contains strings
+  # which identify the old style hash based CWM widgets.
+  # Can be passed to {Yast::CWMClass#ShowAndRun Yast::CWM.ShowAndRun}
+  class StringTerm < Yast::Term; end
+
+  # A {Hash{String=>Object}} that {Yast::CWMClass} knows to handle
```

Comment:
Well, it `is_a? Hash`, which is IMHO expressed well enough by its name, and it is not a generic hash. It is a specific kind of hash that tries to be an object, so I think we are fully justified in documenting it in this way. 

Observe the API docs in CWM.rb where we have several kinds of hashes. I think this does improve the situation.

You are right that it would be good to document the actual members of the hash, but that is too much yak shaving for me now. 

Anyway, what would improve the situation for you and make you less confused?